src/webview/js/rewriteHtml.js

```diff
@@ -56,4 +53,54 @@ const rewriteImageUrls = (auth: Auth, element: Element | Document = document) =>
   });
 };
 
-export default rewriteImageUrls;
+/**
+ * Rewrite the 'time' elements under the given root, inclusive, to make them more readable.
+ *
+ * 1. Changes text to localized human readable date-time.
+ * 2. Adds 'original-text' attribrute, to show original text alert dialog on press.
+ */
+const rewriteTime = (element: Element | Document) => {
+  // Find the time elements to act on.
+  const timeElements = [].concat(
+    element instanceof HTMLTimeElement ? [element] : [],
+    Array.from(element.getElementsByTagName('time')),
+  );
+
+  timeElements.forEach(elem => {
+    // Present only because the Flow DOM libdef does not have a specific definition
+    // for `getElementsByTagName('time')`, so the typedef of `timeElements` is wider
+    // than it actually is.
+    if (!(elem instanceof HTMLTimeElement)) {
+      return;
+    }
+
+    const timeStamp = elem.dateTime;
+    const text = elem.innerText;
+
+    // 'text' is guaranteed to be present, as the server always supplies it.
+    // The conditional is present only to satisfy Flow.
+    if (text == null) {
+      return;
+    }
```

Comment:
Is this conditional actually needed? The code gets a lot shorter if it's taken out -- and simpler to think about, as there's one fewer path through the control flow.

At a quick experiment, I get no Flow errors if I just remove this.